src/gsy_e/gsy_e_core/simulation.py

```diff
@@ -925,7 +925,7 @@ def __init__(self, simulation: Simulation) -> None:
         self.live_events = LiveEvents(simulation.config)
         self.redis_connection = RedisSimulationCommunication(
             simulation_status=simulation.status,
-            simulation_id=simulation._simulation_id,
+            simulation_id=simulation.simulation_id,
```

Comment:
Forget my question about the _status member, I assume it is needed for the same reason as this one.

Reply:
Exactly. For the other reviewers: this PR changes the interface of `SimulationExternalEvents` so that it now accepts just one parameter (`simulation`). This is a refactoring to make the code more OOP oriented, which is our paradigm; if you're curious, you can find some more info in the "Refactoring" book, and a short visual description on the website ([Introduce Parameter Object](https://refactoring.com/catalog/introduceParameterObject.html) and [Preserve Whole Object](https://refactoring.com/catalog/preserveWholeObject.html)).

Because of this, we now access directly some attributes like `Simulation._status` and `Simulation_simulation_id` from another class. So these parameters shouldn't be considered private anymore, but rather public. That's why I removed their underscores.

Reply:
Makes sense, thanks!
One nit though: IMO it makes more sense that the `SimulationExternalEvents` class lives at the top among the other helper classes for the `Simulation` class (as before).

Reply:
The reason why I moved it is to avoid errors with the type hints classes (that need to be declared before they're used). I could probably change it (e.g. using a string). However, I think this class should actually live in a different module. I think you mentioned it in one of your slack messages a few days ago.